internal/bminventory/inventory.go

```diff
@@ -4687,3 +4711,237 @@ func (b *bareMetalInventory) AddOpenshiftVersion(ctx context.Context, ocpRelease
+		Host:                  *host,
+		NextStepRunnerCommand: b.generateV2NextStepRunnerCommand(ctx, &params),
+	}
+
+	if err := tx.Commit().Error; err != nil {
+		log.Error(err)
+		return installer.NewRegisterHostInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	if infraEnv.ClusterID != "" {
+		if err := b.crdUtils.CreateAgentCR(ctx, log, params.NewHostParams.HostID.String(), cluster.KubeKeyNamespace, cluster.KubeKeyName, cluster.ID); err != nil {
+			log.WithError(err).Errorf("Fail to create Agent CR. Namespace: %s, Cluster: %s, HostID: %s", cluster.KubeKeyNamespace, cluster.KubeKeyName, params.NewHostParams.HostID.String())
+			return installer.NewRegisterHostInternalServerError().
+				WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		}
+	}
+
+	txSuccess = true
+
+	return installer.NewV2RegisterHostCreated().WithPayload(&hostRegistration)
+}
+
+func (b *bareMetalInventory) V2GetNextSteps(ctx context.Context, params installer.V2GetNextStepsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var steps models.Steps
+
+	txSuccess := false
+	tx := b.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("get next steps failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("get next steps failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Errorf("failed to start db transaction")
+		return installer.NewUpdateClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, errors.New("DB error, failed to start transaction")))
+	}
+
+	//TODO check the error type
+	host, err := common.GetV2HostFromDB(tx, params.InfraEnvID.String(), params.HostID.String())
+	if err != nil {
+		log.WithError(err).Errorf("failed to find host: %s", params.HostID)
+		return installer.NewGetNextStepsNotFound().
+			WithPayload(common.GenerateError(http.StatusNotFound, err))
+	}
+
+	host.CheckedInAt = strfmt.DateTime(time.Now())
+	if err = tx.Model(&host).UpdateColumn("checked_in_at", host.CheckedInAt).Error; err != nil {
+		log.WithError(err).Errorf("failed to update host: %s", params.HostID.String())
+		return installer.NewV2GetNextStepsInternalServerError()
+	}
+
+	if err = tx.Commit().Error; err != nil {
+		log.Error(err)
+		return installer.NewGetNextStepsInternalServerError()
+	}
+	txSuccess = true
+
+	steps, err = b.hostApi.GetNextSteps(ctx, &host.Host)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get steps for host %s infra-env %s", params.HostID.String(), params.InfraEnvID.String())
+	}
+
+	return installer.NewV2GetNextStepsOK().WithPayload(&steps)
+}
+
+func (b *bareMetalInventory) V2PostStepReply(ctx context.Context, v2Params installer.V2PostStepReplyParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	host, err := common.GetV2HostFromDB(b.db, v2Params.InfraEnvID.String(), v2Params.HostID.String())
```

Comment:
this function should be called GetHostFromDB

Reply:
Whatever we call it, there is another function with the same name that extracts host according to cluster-id

Reply:
see my answer above